cg/meta/observations/balsamic_observations_api.py

```diff
@@ -45,43 +54,122 @@ def loqusdb_sequencing_methods(self) -> list[str]:
         """Return sequencing methods that are eligible for cancer Loqusdb uploads."""
         return LOQUSDB_CANCER_SEQUENCING_METHODS
 
-    def is_analysis_type_eligible_for_observations_upload(self, case_id) -> bool:
+    def is_analysis_type_eligible_for_observations_upload(self, case: Case) -> bool:
         """Return whether the cancer analysis type is eligible for cancer Loqusdb uploads."""
-        if self.analysis_api.is_analysis_normal_only(case_id):
-            LOG.error(f"Normal only analysis {case_id} is not supported for Loqusdb uploads")
+        if self.analysis_api.is_analysis_normal_only(case.internal_id):
+            LOG.error(
+                f"Normal only analysis {case.internal_id} is not supported for Loqusdb uploads"
+            )
+            return False
+        prep_category: str = case.samples[0].prep_category
+        if prep_category in [
+            SeqLibraryPrepCategory.TARGETED_GENOME_SEQUENCING,
+            SeqLibraryPrepCategory.WHOLE_EXOME_SEQUENCING,
+        ] and self._is_paired_analysis(case):
+            LOG.error(
+                f"Paired analysis {case.internal_id} is not supported for TGS Loqusdb uploads"
+            )
             return False
         return True
 
+    @staticmethod
+    def _is_paired_analysis(case: Case) -> bool:
+        return len(case.samples) > 1
+
     def is_case_eligible_for_observations_upload(self, case: Case) -> bool:
         """Return whether a cancer case is eligible for observations upload."""
         return all(
             [
                 self.is_customer_eligible_for_observations_upload(case.customer.internal_id),
-                self.is_sequencing_method_eligible_for_observations_upload(case.internal_id),
-                self.is_analysis_type_eligible_for_observations_upload(case.internal_id),
-                self.is_sample_source_eligible_for_observations_upload(case.internal_id),
+                self.is_analysis_type_eligible_for_observations_upload(case),
+                self.is_sample_source_type_ffpe(case.internal_id),
```

Comment:
Should it not be the reverse? Now it would return True if the sample is FFPE, which will allow the upload, which is the opposite of what we want to do, right?  


```suggestion
                not self.is_sample_source_type_ffpe(case.internal_id),
```